### PR TITLE
3292 colon in discover

### PR DESCRIPTION
**Pull request by alvacouch**
This code modifies the discovery command parser in three ways: 

1. Eliminate error messages for invalid fields in `foo:bar` kinds of queries. Parse invalid keyword entries, e.g., `foo:bar` as literals to be searched for. 
3. Eliminate parsing of `-` and `+` synonyms for `NOT` and include. 

Note that the special keywords `AND`, `OR`, `NOT` are still present and the expressive power of the query engine has not been reduced. Only shorthand syntaxes have been eliminated. 

This makes the strange long names in the CZO use case more easily discoverable. 

### Pull Request Checklist: 
- [x] Positive Test Case Written by Dev

- [x] Automated Testing

- [x] Sufficient User and Developer Documentation

- [ ] Passing Jenkins Build

- [ ] Peer Code review and approval

### Positive Test Case
1. Type discovery request with colons and invalid keyword, e.g., `foo:bar`. Error is not returned. Instead, this is searched for literally. 
2. Type a discovery request `-couch` or `+couch`. This should be interpreted literally, and not as `NOT couch`.

### CODE REVIEW

**Comment by alvacouch** [comment]
@dtarb @martinseul @Lizabrazil A meta-comment: If we really want to have users to be able to put _anything_ in a title, then we need to support this by making the query language as different from anything that can appear in a title as possible. Thus, there is some argument for making all the keywords _ALL CAPS_, e.g. `AUTHOR:`, `TITLE:`, on the grounds that most people don't title things with _"lines of yelling"_ (as the spam filters call the use of all-caps).  In other words, allowing both `author:` and `AUTHOR:` is the _wrong_ direction.

**Comment by alvacouch** [comment]
@Lizabrazil @martinseul @cuahsimarko @dtarb 
The change has been made to eliminate lower-casing of keywords and this is ready to merge.

File: hs_core/discovery_parser.py [hunk 1]
```diff
@@ -78,9 +78,8 @@ class ParseSQ(object):
     OP = {
         'AND': operator.and_,
         'OR': operator.or_,
-        'NOT': operator.inv,
-        '+': operator.and_,
-        '-': operator.inv,
+        'NOT': operator.inv,  # alias '-' removed 4/5/2019
+        # alias '+' (for include) removed 4/5/2019
```
Comment (sblack-usu) [inline hs_core/discovery_parser.py]: I'm not a fan of commenting out code to remove it.  The git commit history documents this for us without cluttering our code.  I am goo with putting in a comment that explains why + and - were removed intentionally.

File: hs_core/tests/api/native/test_discovery_parser.py [hunk 2]
```diff
@@ -77,14 +76,14 @@ def test_parse_with_new_default(self):
     def test_operators(self):
         testcase = {
             "note": str(SQ(content="note")),
-            "need -note": str(SQ(content="need") & ~SQ(content="note")),
-            "need +note": str(SQ(content="need") & SQ(content="note")),
-            "need+note": str(SQ(content="need+note")),
-            "iphone AND NOT subject:10": str(SQ(content="iphone") & ~SQ(
-                subject="10")),
+            # removed '-' 4/5/2019 "need -note": str(SQ(content="need") & ~SQ(content="note")),
```
Comment (sblack-usu) [inline hs_core/tests/api/native/test_discovery_parser.py]: Instead of removing these, could you add a test that tests the new outcome?

Comment (cuahsimarko) [reply hs_core/tests/api/native/test_discovery_parser.py]: PyTest!

Comment (cuahsimarko) [reply hs_core/tests/api/native/test_discovery_parser.py]: Maybe @zhiyuli , Mahesh or I can look into that or you can do a quick nosetest.

File: hs_core/discovery_parser.py [hunk 3]
```diff
@@ -219,9 +220,11 @@ def handle_field_query(self):
         search_field = mat.group(1)
         search_operator = mat.group(2)
         if search_field not in self.KNOWN_FIELDS:
-            raise FieldNotRecognizedError(
-                "Field delimiter '{}' is not recognized."
-                .format(search_field))
+            self.handle_normal_query()
+            return
+            # raise FieldNotRecognizedError(
```
Comment (alvacouch) [inline hs_core/discovery_parser.py]: This is a temporary fix and I would like to remember what used to happen.

**Comment by alvacouch** [comment]
@sblack-usu I think this is ready now for merge.

**Comment by alvacouch** [comment]
@cuahsimarko This is approved as text but not in git. Please approve in git and merge. (cc: @martinseul )

**Pull request merged**